From 6fbd388fb8c08a14b6e09685ebd969ac178120d8 Mon Sep 17 00:00:00 2001
From: Pawel Furtak <pawel.furtak@intel.com>
Date: Sun, 25 Nov 2018 16:58:08 +0100
Subject: [PATCH 208/292] ASoC: Intel: Skylake: Export skylake functions for
 virtio.

In order to reuse existing implementation in virtualized
system, additional functions have to be exported.

Change-Id: I3c6af0e9a51894073f8dcef3ba4a4aed8fa56386
Signed-off-by: Pawel Furtak <pawel.furtak@intel.com>
Reviewed-on:
Reviewed-by: Rojewski, Cezary <cezary.rojewski@intel.com>
Tested-by: Rojewski, Cezary <cezary.rojewski@intel.com>
---
 sound/soc/intel/skylake/skl-messages.c  |  1 +
 sound/soc/intel/skylake/skl-pcm.c       | 15 ++++++++++-----
 sound/soc/intel/skylake/skl-sst-utils.c |  2 ++
 sound/soc/intel/skylake/skl.h           |  7 +++++++
 4 files changed, 20 insertions(+), 5 deletions(-)

diff --git a/sound/soc/intel/skylake/skl-messages.c b/sound/soc/intel/skylake/skl-messages.c
index 4fa273358ee3..ac3405de318a 100644
--- a/sound/soc/intel/skylake/skl-messages.c
+++ b/sound/soc/intel/skylake/skl-messages.c
@@ -335,6 +335,7 @@ const struct skl_dsp_ops *skl_get_dsp_ops(int pci_id)
 
 	return NULL;
 }
+EXPORT_SYMBOL(skl_get_dsp_ops);
 
 int skl_init_dsp(struct skl *skl)
 {
diff --git a/sound/soc/intel/skylake/skl-pcm.c b/sound/soc/intel/skylake/skl-pcm.c
index bf2ad1c5be01..e291281817c6 100644
--- a/sound/soc/intel/skylake/skl-pcm.c
+++ b/sound/soc/intel/skylake/skl-pcm.c
@@ -1450,7 +1450,7 @@ int skl_dai_load(struct snd_soc_component *cmp, int index,
 	return 0;
 }
 
-static int skl_platform_open(struct snd_pcm_substream *substream)
+int skl_platform_open(struct snd_pcm_substream *substream)
 {
 	struct snd_soc_pcm_runtime *rtd = substream->private_data;
 	struct snd_soc_dai_link *dai_link = rtd->dai_link;
@@ -1462,6 +1462,7 @@ static int skl_platform_open(struct snd_pcm_substream *substream)
 
 	return 0;
 }
+EXPORT_SYMBOL(skl_platform_open);
 
 static int skl_coupled_trigger(struct snd_pcm_substream *substream,
 					int cmd)
@@ -1536,7 +1537,7 @@ static int skl_coupled_trigger(struct snd_pcm_substream *substream,
 	return 0;
 }
 
-static int skl_platform_pcm_trigger(struct snd_pcm_substream *substream,
+int skl_platform_pcm_trigger(struct snd_pcm_substream *substream,
 					int cmd)
 {
 	struct hdac_bus *bus = get_bus_ctx(substream);
@@ -1546,6 +1547,7 @@ static int skl_platform_pcm_trigger(struct snd_pcm_substream *substream,
 
 	return 0;
 }
+EXPORT_SYMBOL(skl_platform_pcm_trigger);
 
 /* update SPIB register with appl position */
 static int skl_platform_ack(struct snd_pcm_substream *substream)
@@ -1682,14 +1684,15 @@ static const struct snd_pcm_ops skl_platform_ops = {
 	.ack = skl_platform_ack,
 };
 
-static void skl_pcm_free(struct snd_pcm *pcm)
+void skl_pcm_free(struct snd_pcm *pcm)
 {
 	snd_pcm_lib_preallocate_free_for_all(pcm);
 }
+EXPORT_SYMBOL(skl_pcm_free);
 
 #define MAX_PREALLOC_SIZE	(32 * 1024 * 1024)
 
-static int skl_pcm_new(struct snd_soc_pcm_runtime *rtd)
+int skl_pcm_new(struct snd_soc_pcm_runtime *rtd)
 {
 	struct snd_soc_dai *dai = rtd->cpu_dai;
 	struct hdac_bus *bus = dev_get_drvdata(dai->dev);
@@ -1716,6 +1719,7 @@ static int skl_pcm_new(struct snd_soc_pcm_runtime *rtd)
 
 	return retval;
 }
+EXPORT_SYMBOL(skl_pcm_new);
 
 static int skl_get_module_info(struct skl *skl, struct skl_module_cfg *mconfig)
 {
@@ -1835,7 +1839,7 @@ static int skl_get_probe_widget(struct snd_soc_component *component,
 	return 0;
 }
 
-static int skl_platform_soc_probe(struct snd_soc_component *component)
+int skl_platform_soc_probe(struct snd_soc_component *component)
 {
 	struct hdac_bus *bus = dev_get_drvdata(component->dev);
 	struct skl *skl = bus_to_skl(bus);
@@ -1904,6 +1908,7 @@ static int skl_platform_soc_probe(struct snd_soc_component *component)
 
 	return 0;
 }
+EXPORT_SYMBOL(skl_platform_soc_probe);
 
 static const char* const dsp_log_text[] =
 	{"QUIET", "CRITICAL", "HIGH", "MEDIUM", "LOW", "VERBOSE"};
diff --git a/sound/soc/intel/skylake/skl-sst-utils.c b/sound/soc/intel/skylake/skl-sst-utils.c
index 006da1639cd1..de3fc96607b3 100644
--- a/sound/soc/intel/skylake/skl-sst-utils.c
+++ b/sound/soc/intel/skylake/skl-sst-utils.c
@@ -588,6 +588,7 @@ int snd_skl_parse_uuids(struct sst_dsp *ctx, const struct firmware *fw,
 	skl_freeup_uuid_list(skl);
 	return ret;
 }
+EXPORT_SYMBOL(snd_skl_parse_uuids);
 
 static int skl_parse_hw_config_info(struct sst_dsp *ctx, u8 *src, int limit)
 {
@@ -799,6 +800,7 @@ int skl_prepare_lib_load(struct skl_sst *skl, struct skl_lib_info *linfo,
 
 	return 0;
 }
+EXPORT_SYMBOL(skl_prepare_lib_load);
 
 void skl_release_library(struct skl_lib_info *linfo, int lib_count)
 {
diff --git a/sound/soc/intel/skylake/skl.h b/sound/soc/intel/skylake/skl.h
index f0e24328f58b..dc02c93ef727 100644
--- a/sound/soc/intel/skylake/skl.h
+++ b/sound/soc/intel/skylake/skl.h
@@ -224,6 +224,13 @@ int skl_dsp_set_dma_control(struct skl_sst *ctx, u32 *caps,
 
 struct skl_module_cfg;
 
+int skl_pcm_new(struct snd_soc_pcm_runtime *rtd);
+void skl_pcm_free(struct snd_pcm *pcm);
+int skl_platform_soc_probe(struct snd_soc_component *component);
+int skl_platform_open(struct snd_pcm_substream *substream);
+int skl_platform_pcm_trigger(struct snd_pcm_substream *substream,
+	int cmd);
+
 #ifdef CONFIG_DEBUG_FS
 struct skl_debug *skl_debugfs_init(struct skl *skl);
 void skl_debug_init_module(struct skl_debug *d,
-- 
2.21.0

